File: 3227-find-missing-and-repeated-values/find-missing-and-repeated-values.java
class Solution {
    public int[] findMissingAndRepeatedValues(int[][] grid) {
        Set<Integer> forA = new HashSet<>();
        int a = 0;
        int n = grid.length;
        long actualSum = 0;
        long expectedSum = (long) n * n * (n * n + 1) / 2;

        for (int i = 0; i < grid.length; i++) {
            for (int j = 0; j < grid[0].length; j++) {
                actualSum += grid[i][j];
                if (forA.contains(grid[i][j])) {
                    a = grid[i][j];
                }
                forA.add(grid[i][j]);
            }
        }

        long b = expectedSum - actualSum + a;
        return new int[]{a, (int) b};
    }
}

File: 3227-find-missing-and-repeated-values/find-missing-and-repeated-values.py
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        for_a = set()
        a = 0
        n = len(grid)
        actual_sum = 0
        expected_sum = n * n * (n * n + 1) // 2

        for i in range(len(grid)):
            for j in range(len(grid[0])):
                actual_sum += grid[i][j]
                if grid[i][j] in for_a:
                    a = grid[i][j]
                for_a.add(grid[i][j])

        b = expected_sum - actual_sum + a
        return [a, b]